shuffle/move/examples/main/devapi.ts

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: Apache-2.0
+
+// Generated on new project creation. Invoked by shuffle CLI.
+
+// Creates typescript wrappers around the Developer API for easier consumption,
+// including endpoints: ledgerInfo, resources, modules, and some of transactions.
+// Developer API: https://docs.google.com/document/d/1KEPnGGU3zg_RmN8V4r2ms_MFPwsTMNyK7jCUFygviDg/edit#heading=h.hesw425dw9gz
+
+// deno-lint-ignore-file no-explicit-any
+import { createRemote } from "https://deno.land/x/gentle_rpc@v3.1/mod.ts";
+import * as context from "./context.ts";
+
+console.log(await ledgerInfo());
+
+export async function ledgerInfo() {
+  const res = await fetch(context.relativeUrl("/"));
+  return await res.json();
+}
+
+export async function transactions() {
+  const res = await fetch(context.relativeUrl("/transactions"));
+  return await res.json();
+}
+
+export async function accountTransactions() {
+  const remote = createRemote("http://127.0.0.1:8080/v1");
+  return await remote.call(
+    "get_account_transactions",
+    [context.senderAddress.substring(2), 0, 10, true],
+  );
+}
+
+// deno-lint-ignore ban-types
+export async function resources(addr?: string): Promise<object[]> {
+  addr = context.addressOrDefault(addr);
+  const res = await fetch(context.relativeUrl(`/accounts/${addr}/resources`));
+  return await res.json();
+}
+
+export async function modules(addr?: string) {
+  addr = context.addressOrDefault(addr);
+  const res = await fetch(context.relativeUrl(`/accounts/${addr}/modules`));
+  return await res.json();
+}
+
+// Gets the sender address's account resource from the developer API.
+// Example payload below:
+// {
+//   "type": "0x1::DiemAccount::DiemAccount",
+//   "value": {
+//     "sequence_number": "2",
+export async function account(addr?: string) {
+  addr = context.addressOrDefault(addr);
+  const res = await resources(addr);
+  return res
+    .find(
+      (entry: any) => entry["type"] == "0x1::DiemAccount::DiemAccount",
+    );
+}
+
+export async function sequenceNumber(addr?: string): Promise<number> {
+  const acc = await account(addr);
+  if (acc) {
+    return parseInt(acc["value"]["sequence_number"]);
+  }
+  throw "unable to find account";
+}
+
+export async function accounts() {
+  return [await account(undefined)];
+}
+
+export async function postTransactionBcs(
+  body: string | Uint8Array,
+): Promise<any> {
+  const settings = {
+    method: "POST",
+    body: body,
+    headers: {
+      "Content-Type": "application/vnd.bcs+signed_transaction",
+    },
+  };
+  const res = await fetch(context.relativeUrl("/transactions"), settings);
+  return await res.json();
+}
+
+export async function postTransactionSigningMessage(
+  body: string,
+): Promise<any> {
+  const resp = await fetch(
+    context.relativeUrl("/transactions/signing_message"),
+    {
+      method: "POST",
+      body: body,
+      headers: {
+        "Content-Type": "application/json",
+      },
+    },
+  );
+
+  if (!isSuccess(resp.status)) {
```

Comment:
Looks like you want same error handling logic, could we wrap these logic into 'fetch' function wrapper? As above resources and modules functions don't handle errors